### PR TITLE
React Project - Calculator , MILESTONE 2 

### PR DESCRIPTION
**Pull request by tanzila-abedin**
**4 following components  are created to render In APP:**
 
1. Wrapper: Parent component of Display and ButtonPanel.
2. Display: child of wrapper component.
3. ButtonPanel: Parent of Button component
4. Button: Child of ButtonPanel.

- Individual files were created for each component.
- 4 functional components were created.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,27 +1,39 @@
-import logo from './logo.svg';
```
Comment (eliasecasta) [inline src/App.js:1]: - [x] Your App.js file should be placed inside your components folder to comply with the project requirements, please do so 👍

File: src/App.js [hunk 2]
```diff
@@ -1,27 +1,39 @@
...
-import './App.css';
+// import './App.css';
```
Comment (eliasecasta) [inline src/App.js]: - [x] Please remove this and any other commented code that you may have to keep your codebase clean.

File: src/App.js [hunk 3]
```diff
@@ -1,27 +1,39 @@
...
+  ['C', '+-', '%', '/'],
+  [7, 8, 9, 'X'],
+  [4, 5, 6, '-'],
+  [1, 2, 3, '+'],
+  [0, '.', '='],
+];
 
 function App() {
   return (
     <div className="App">
```
Comment (eliasecasta) [inline src/App.js]: - [x] The React.fragment short syntax should be used to wrap the children components for this function, please do so to comply with the project requirements: Has children components wrapped in [React.Fragment](https://reactjs.org/docs/fragments.html#short-syntax) short syntax ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/app/ReactFragment.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896092/))

File: src/components/Display.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import PropTypes from 'prop-types';
+import { Textfit } from 'react-textfit';
+// import '../App.css';
+
+const Display = ({ value }) => (
+  <Textfit className="display" mode="single" max={70}>
+    {value}
```
Comment (eliasecasta) [inline src/components/Display.js:7]: - [x] its result prop should be a String with default value = 0 (Right now, it is only being initialized with the value 0 but 0 is not set as the default value, you can do so by equaling value to 0 at the function parameter)  ([should be String screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/validateAsString.png)) ([default props screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/DefaultProp.png))

File: src/App.js [hunk 5]
```diff
@@ -1,27 +1,39 @@
...
+      <Wrapper>
+        <Display value="0" />
+        <ButtonPanel>
+          {btnValues.flat().map((btn) => (
+            <Button
+              key={btn}
+              className={btn === '=' ? 'equals' : ''}
+              value={String(btn)}
+              onClick={() => {
+                console.log(`${btn} clicked!`);
```
Comment (eliasecasta) [inline src/App.js]: - [x] Please avoid the use of console.log in code being reviewed, you are free to use it while developing but it should not be present in the pull request, plus this is giving warning errors when running your app, please remove it for something else 👍

**Review by eliasecasta — CHANGES_REQUESTED** [review]
# Status:  Changes Required ♻️
_Check the comments under the review_
<hr>

Hello @tanzila-abedin, I am Elias, your reviewer for this PR

Well done so far with your project!

I really liked the clever way that you managed the button rendering, great job so far! You are missing some really small details before moving on, good luck!

I will mention to you a small summary of things that need to be changed in order to keep moving:

- [x] Please check the comments that I left in this PR.

Please correct those issues and send another review request, good luck, and well done so far! 

Code review by Elias Castañeda

_(Feel free to leave any questions or comments in the PR thread if something is not 100% clear)_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/App.css [hunk 6]
```diff
@@ -1,39 +1,56 @@
...
-    transform: rotate(360deg);
-  }
+.equals {
+  grid-column: 3 / 5;
+  background-color: rgb(243, 61, 29);
 }
+
+.equals:hover {
+  background-color: rgb(228, 39, 15);
+} */
```
Comment (george-shammar) [inline src/App.css:56]: - [ ] As mentioned by the last reviewer, please clean up commented codes from your codebase. All the codes in this file have been commented out. Please, let's clean it up. :pray: You may delete the file altogether if its not useful.

**Review by george-shammar — CHANGES_REQUESTED** [review]
Hi @tanzila-abedin  

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Pull request merged**